Hackathon Guide

sploiselle opened the pull request:
Wanted to put together a resource for Hackathon students to get them started with using CockroachDB in Node.js (by far the most popular framework for hacks). This will be the content we point students to when they come up and talk to us.

My experience at two hackathons last year leads me to believe we need this because a lot of students struggle with parsing through the documentation and finding all of the bits and pieces they need. All of this content exists elsewhere, but by aggregating everything into a single place, we're table to ensure they don't fall off the "happy path," and if they do, we have many points that we can put them back on.

### Questions

Sending this up to see what @jseldess and @jess-edwards think. A few questions for y'all:

- Does this work on the docs site, or should it be a blog post?
- If the docs site, there are questions to answer about its location. I think it might be fine not listed on the side bar, but would be glad to defer to whatever Jesse thinks is best.
- If the blog, do we want to spruce it up?

As this is a WIP there is some rough copy/need for better IA (and probably some diagrams).

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/1840)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0a819b0a97605ded46e8c03db08b88ebd43b4bdf/

jseldess commented:
@sploiselle, this is amazing! I definitely think this can and should live on the docs site. For the short-term, we can just not list it in the side-nav (in fact, we have the choice to use a different layout that hides the sidenav, if you think that's a better UX for the hackathon, so users don't click away...). But in the long-term, we should definitely think about how to best expose this. 

In Q4, we'll collaborate with Design to find ways to increase the docs "space". As you've mentioned elsewhere, we should probably have a separate area for Architecture. We may also want a separate area for something like this. It may even be possible to present this in a way that it's not only for hackathons (but still works well for hackathons). Something like rethink's ["10-minute guide"](https://www.rethinkdb.com/docs/guide/javascript/).

Interested in @jess-edwards' thoughts, of course. And @sploiselle, I'm happy to do a content review when you're ready, too. Just let me know.

sploiselle commented:
@jseldess I'd be really into disabling the sidebar for this (for the first run to see how it goes). So glad you think this can live here, but we'll wait and see what Jess says before finalizing.

I've never seen that Rethink guide before––I think that's pretty compelling!

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3b4cdd1a218e988a1c329d91d8d872a31bde9b7d/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/918e2ae242834fcda612f6e54b8ea3e9a3b4a074/

jseldess commented:
@sploiselle, successfully removed the sidenav and got the body to center on the screen, with a little help from @benesch, aka the Universal Wizard.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f9d68e3831980d527dbca6c8d29a16d38627316a/

sploiselle commented:
@justinj Any chance I could get you to give this a once-over to make sure it's technically sound?

justinj commented:
this looks pretty sensible to me generally! I think there will definitely be stuff we'll have to help people with (`JOIN`s in particular I think are tricky enough that a document of this scope would have trouble explaining them in full) but that's why we're there!

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/8a5e143a632bfb29da49dd45556cd5be82de6fb3/

sploiselle commented:
@jseldess Could I get a copy edit?

jseldess commented:
Will review soon, @sploiselle.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+## Quick Start: Code Sample Repo
+
+To get you started quickly, Cockroach Labs offers [a simple Express-based Node.js app on GitHub](https://github.com/sploiselle/bankofroach).
```
Not important for this hackathon, but eventually, it'd be great to move this to a repo in the cockroachdb github account.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+To use the app, you'll still need to install and start `cockroach`, but if you've already done that process, this will make it easy to get started.
+
+<a href="https://github.com/sploiselle/bankofroach" type="button" class="btn btn-primary" style="color:#FFF;">Get the code sample</a>
```
We should style this button like we do others. Right now, the button starts like this:

![screen shot 2017-09-06 at 3 13 32 pm](https://user-images.githubusercontent.com/15893490/30130068-013f7936-9316-11e7-958e-1abdd6164e1e.png)

Then on hover, it looks like this:

![screen shot 2017-09-06 at 3 13 40 pm](https://user-images.githubusercontent.com/15893490/30130077-0a171e42-9316-11e7-872c-25ebccd6e173.png)

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+<div id="toc"></div>
+
+## Quick Start: Code Sample Repo
```
I'm a little confused about how this "quick start" ties into the rest of the guide. If starting cockroach is a prerequisite, would it make sense to move this after that, perhaps as a step?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+- **Locally** on a single machine. This is useful to get started, but can be challenging because only one person can access the database at a time.
+- **Remotely** on a cloud hosting provider. This option can give everyone on your team access to the database, but can be more difficult to configure because you have to deal with your VM's networking rules.
+
+<script>
```
Eventually, we should move this javascript into a separate file and include it on a page via a front-matter field or something. Not important for now, though.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+    sql:        postgresql://root@localhost:26257?sslmode=disable
+    logs:       /Users/me/cockroach-data/logs
+    store[0]:   path=/Users/you/cockroach-data
+    status:     started new node
```
nit: This should be `initialized new cluster`

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+3. Columns, which represent types of data
+4. Rows, which represent individual "observations" of the types of data
+
+This maps really clearly to how you think of object-oriented programming.
```
nit: End with a colon.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+By creating this relationship you can easily do things like find all of a user's orders, or do more complicated things like filter `order` data based on data that exists only in the `customers` table with a `JOIN` (more on that later).
+
+SQL offers an explicit `FOREIGN KEY` constraint, which can ensure your data remains consistent. However, it's also possible to 
```
This sentence ends mid-stream...

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+{{site.data.alerts.callout_success}}<strong>tl;dr</strong>: To relate two tables together use a <code>FOREIGN KEY</code>, which represents a column from one table in another.{{site.data.alerts.end}}
+
+Though this typically works itself out with a solid object-oriented design, it's important to understand tables that you want to relate with one another and ensure that you include columns that will let you do that.
```
Don't think the first clause of this sentence adds much. I'd just start with "It's important to understand...". But I don't feel strongly.

jseldess commented:
This is looking great. One overall question I have is: Should we link to specific docs more often? For example, for specific sql statements? Or do we not want them to leave this page?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+2. Create tables for your data:
+
+    ~~~ sql
+    > CREATE TABLE tbl_name (
```
Maybe replace `TYPE` with actual types, since this is just an example?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+    );
+    ~~~
+
+    Note that we denote which column we want to use to uniquely identify the column by identifying it as the `PRIMARY KEY`.
```
nit: `to uniquely identify rows...`?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+ For some of your tables, you'll want to include a `FOREIGN KEY` columns. You don't have to formally enforce the `FOREIGN KEY` relationship, but you can to ensure you don't end up with inconsistent data:
+
+ ~~~ sql
```
Need to add prompts before each statement. Might be nice to split them into separate code blocks and add "copy-to-clipboard" buttons.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+CockroachDB offers typical "CRUD" operations using standard SQL keywords:
+
+CRUD Operation | SQL Keyword
```
Maybe it's worth adding `UPSERT`, too?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+Below are some quick examples of what these statements look like (and you can find more of them in our [documentation](sql-statements.html)).
+
+#### `INSERT` Example
```
nit: Examples

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+INSERT INTO tbl_name (id, val) VALUES (1, 'a'), (2, 'b'), (3, 'a');
+~~~
+
+#### `SELECT` Example
```
nit: Examples

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+UPDATE tbl_name SET col = 'newVal' WHERE col = 'oldVal';
+~~~
+
+#### `DELETE` Example
```
nit: Examples

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+#### `INSERT` Example
+
+Implicit Columns:
```
nit: Make capitalization consistent with later examples: `Implicit columns:`

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+INSERT INTO tbl_name VALUES (1, 'a'), (2, 'b'), (3, 'a');
+~~~
+
+Explicit Columns:
```
Same as above: `Explicit columns:`

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+SELECT id, val FROM tbl_name;
+~~~
+
+Select specific rows
```
Add a colon.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+## 5. Install pg-promise
+
+To make your code as simple as possible, we recommend using `pg-promise`. This is a Postgres driver that supports using Node.js promises––but with ES7, also lets us `async`/`await`, which makes our code _really_ easy to write and read.
```
Link to pg-promise site?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+Here's a simple example that will show you all of the customers who have data in the `order` table (i.e., all of the customers who have placed an order):
+
+~~~ sql
+SELECT customers.id, customers.name
```
Add prompt and copy-to-clipboard?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+Add a column:
+
+~~~ sql
+ALTER TABLE tbl_name ADD COLUMN foo TYPE;
```
Add prompt and copy-to-clipboard?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+Drop a column:
+
+~~~ sql
+ALTER TABLE tbl_name DROP COLUMN foo;
```
Add prompt and copy-to-clipboard?

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+Here's a quick code sample:
+
+~~~
```
Need to add syntax highlighting.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+Looking at the code above, you can infer how transactions work, but here's what's explicitly happening:
+
+- Transaction are initiated with a `BEGIN` statement
```
Add a period.

jseldess commented on hackathons.md:
```diff
@@ -0,0 +1,563 @@
+
+- Transaction are initiated with a `BEGIN` statement
+- Statements received after `BEGIN` are considered part of the transaction.
+- Once the group of statements are done, end the transaction with the `COMMIT` statement.
```
nit: group ... is done

sploiselle commented:
@jseldess In principal, I don't want them leaving this page too often because it's a little hidden and has most of the relevant content inline. If there are specific instances where we think it would do a disservice to the usability of this page to include info, glad to reshape.

---

Review status: 0 of 5 files reviewed at latest revision, 29 unresolved discussions.

---

*[hackathons.md, line 11 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r1-11:-Ku0fVEPoHatVg0xayCX:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f3babc2e547273747c6a293b00c2b9cbb5454564/hackathons.md#L11)):*
<details><summary><i>Previously, jess-edwards wrote…</i></summary><blockquote>

@sploiselle Could this section some kind of "before you begin" or "pre-requisite" header? Looks like that's your standard across other docs pages.
</blockquote></details>

Done.

---

*[hackathons.md, line 18 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r1-18:-Ku0fxcnWjmdw1DdVkY0:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f3babc2e547273747c6a293b00c2b9cbb5454564/hackathons.md#L18)):*
<details><summary><i>Previously, jess-edwards wrote…</i></summary><blockquote>

@sploiselle could add: hit us on slack / gitter
</blockquote></details>

Done.

---

*[hackathons.md, line 33 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r1-33:-Ku0k3j3A9n0uIcY-K5-:bgdruvz) ([raw file](https://github.com/cockroachdb/docs/blob/f3babc2e547273747c6a293b00c2b9cbb5454564/hackathons.md#L33)):*
<details><summary><i>Previously, jess-edwards wrote…</i></summary><blockquote>

@sploiselle 

For "Locally: Step 1" and "Remotely: Step 2"  - Can you add conversion tracking to the click-to-copy buttons and control-c actions? You can use the same events as are used on the install page. I can look at this with you if you'd like.
</blockquote></details>

Meant to have you show me this before the hackathon. Don't think I'm going to be able to figure it out before, but can definitely add it to the page.

---

*[hackathons.md, line 134 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r1-134:-Ku0kVfEra1MqQqmlLSD:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f3babc2e547273747c6a293b00c2b9cbb5454564/hackathons.md#L134)):*
<details><summary><i>Previously, jess-edwards wrote…</i></summary><blockquote>

@sploiselle  "that the URL that displays"
</blockquote></details>

Done.

---

*[hackathons.md, line 23 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-23:-Ku0ga4iZzoefmy_ttKk:b44pgzq) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L23)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I'm a little confused about how this "quick start" ties into the rest of the guide. If starting cockroach is a prerequisite, would it make sense to move this after that, perhaps as a step?
</blockquote></details>

Moved to a 2.5 header

---

*[hackathons.md, line 25 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-25:-Ku0gjYIOlehEMdBQxdB:b30vvih) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L25)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Not important for this hackathon, but eventually, it'd be great to move this to a repo in the cockroachdb github account.
</blockquote></details>

Avoided that for the time being to minimize the shame of writing embarrassing code :)

---

*[hackathons.md, line 29 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-29:-Ku0jnrGGBuv55a7xaI0:b9wwon9) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L29)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

We should style this button like we do others. Right now, the button starts like this:

![screen shot 2017-09-06 at 3 13 32 pm](https://user-images.githubusercontent.com/15893490/30130068-013f7936-9316-11e7-958e-1abdd6164e1e.png)

Then on hover, it looks like this:

![screen shot 2017-09-06 at 3 13 40 pm](https://user-images.githubusercontent.com/15893490/30130077-0a171e42-9316-11e7-872c-25ebccd6e173.png)



</blockquote></details>

Fixed! Wasn't doing it earlier, but I'm repoing it now.

---

*[hackathons.md, line 38 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-38:-Ku0jtBVDXc0ZnrAyUxn:b-zggnmq) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L38)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Eventually, we should move this javascript into a separate file and include it on a page via a front-matter field or something. Not important for now, though.
</blockquote></details>

Agreed. This and the install page could both use some cleanup.

---

*[hackathons.md, line 188 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-188:-Ku0tRkeNBTeiVrNgibj:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L188)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: This should be `initialized new cluster`
</blockquote></details>

Done.

---

*[hackathons.md, line 239 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-239:-Ku0ka9zPWqGFSc_hRuK:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L239)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: End with a colon.
</blockquote></details>

Done.

---

*[hackathons.md, line 276 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-276:-Ku0kizfvSqof636pPLe:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L276)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Don't think the first clause of this sentence adds much. I'd just start with "It's important to understand...". But I don't feel strongly. 
</blockquote></details>

Done.

---

*[hackathons.md, line 282 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-282:-Ku0ktYAeORDvDzDJPKf:b-2mj6qm) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L282)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

This sentence ends mid-stream...
</blockquote></details>

Typo! Fixed.

---

*[hackathons.md, line 305 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-305:-Ku0kzLQj4fyA0XPF_nP:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L305)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Maybe replace `TYPE` with actual types, since this is just an example?
</blockquote></details>

Done.

---

*[hackathons.md, line 312 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-312:-Ku0l9QfvRSAVj3ira9i:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L312)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `to uniquely identify rows...`?
</blockquote></details>

Done.

---

*[hackathons.md, line 320 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-320:-Ku0tSt6SyWGzJwiuiXe:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L320)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Need to add prompts before each statement. Might be nice to split them into separate code blocks and add "copy-to-clipboard" buttons.
</blockquote></details>

Done.

---

*[hackathons.md, line 345 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-345:-Ku0nd2ohzRpcwW8TxAr:b-y5kws2) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L345)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Maybe it's worth adding `UPSERT`, too?
</blockquote></details>

Figured that was more advanced. Definitely worth keeping in mind, but don't think I have time to add it before the hackathon.

---

*[hackathons.md, line 354 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-354:-Ku0sQkn_DsNY5Ke_SL7:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L354)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Examples
</blockquote></details>

Done.

---

*[hackathons.md, line 356 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-356:-Ku0sginIcFFIUhzUcHd:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L356)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Make capitalization consistent with later examples: `Implicit columns:`
</blockquote></details>

Done.

---

*[hackathons.md, line 362 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-362:-Ku0shX0mgyWXSAn7kzA:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L362)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Same as above: `Explicit columns:`
</blockquote></details>

Done.

---

*[hackathons.md, line 368 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-368:-Ku0shvWBcEeaZbJK9YA:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L368)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Examples
</blockquote></details>

Done.

---

*[hackathons.md, line 382 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-382:-Ku0smFhH3m8x8kOTgQT:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L382)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add a colon.
</blockquote></details>

Done.

---

*[hackathons.md, line 394 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-394:-Ku0smt-WMTB0D5nvjkl:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L394)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: Examples
</blockquote></details>

Done.

---

*[hackathons.md, line 414 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-414:-Ku0szJ2r-SBhBY0446V:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L414)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Link to pg-promise site?
</blockquote></details>

Done.

---

*[hackathons.md, line 502 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-502:-Ku0t9AGdxqKkgODkiO9:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L502)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add prompt and copy-to-clipboard?
</blockquote></details>

Done.

---

*[hackathons.md, line 515 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-515:-Ku0tACIpjRdFC4CNbox:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L515)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add prompt and copy-to-clipboard?
</blockquote></details>

Done.

---

*[hackathons.md, line 521 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-521:-Ku0tAdSxSe0lJgPaHcn:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L521)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add prompt and copy-to-clipboard?
</blockquote></details>

Done.

---

*[hackathons.md, line 532 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-532:-Ku0tH4PAbNFL49LKWUc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L532)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Need to add syntax highlighting.
</blockquote></details>

Done.

---

*[hackathons.md, line 554 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-554:-Ku0tNr25ZyoC3ZT24u0:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L554)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add a period.
</blockquote></details>

Done.

---

*[hackathons.md, line 556 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-556:-Ku0tQM9q0QHSZft-tIa:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L556)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: group ... is done
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-:-Ku0ew45Y6Rbl6zB8UOG:b3vvsc1)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/839ea9b0d75bbf88487793fcf6c9f5c6ab7b3d26/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5c74b02d6f0544a7dbee4935a4cf0283b5c9172e/

jseldess commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-:-Ku16iXgQM_FSTMK2FT5:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[hackathons.md, line 29 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Krp672ouAgy-CKZZ-vf-r3-29:-Ku14996ZRG_9NcnyFU4:b7xzzki) ([raw file](https://github.com/cockroachdb/docs/blob/8a5e143a632bfb29da49dd45556cd5be82de6fb3/hackathons.md#L29)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary><blockquote>

Fixed! Wasn't doing it earlier, but I'm repoing it now.
</blockquote></details>

Think it'd be nice to style this button like the other buttons on the site. But it's working for me on hover now at least.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1840)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[hackathons.md, line 138 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Ku14mAu9v3PMQntRB7B:-Ku14mAu9v3PMQntRB7C:bh5qgf0) ([raw file](https://github.com/cockroachdb/docs/blob/5c74b02d6f0544a7dbee4935a4cf0283b5c9172e/hackathons.md#L138)):*
> ```Markdown
> 
>     <li>
>       <p>Download the <a href="https://binaries.cockroachdb.com/cockroach-{{page.release_info.version}}.linux-amd64.tgz">CockroachDB archive</a> for OS X, and extract the binary:</p>
> ```

Think this should be `for Linux`, not `for OS X`...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1840)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[hackathons.md, line 161 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Ku150rIj4K6w_atjZ-5:-Ku150rIj4K6w_atjZ-6:b-xh7ipu) ([raw file](https://github.com/cockroachdb/docs/blob/5c74b02d6f0544a7dbee4935a4cf0283b5c9172e/hackathons.md#L161)):*
> ```Markdown
>       <p>You should see an output that looks somewhat like this:</p>
> 
>       <pre class="highlight"><code>Build Tag:    v1.0.3
> ```

nit: In the download urls, you're using `{{ page.release_info.version }}` to get the most recent stable version, which is 1.0.6, whereas here you're showing v1.0.3. Might as well use the liquid tag here, too? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1840)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[hackathons.md, line 186 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1840#-Ku15JF3KNRGrpqns_rR:-Ku15JF3KNRGrpqns_rS:b2uih3n) ([raw file](https://github.com/cockroachdb/docs/blob/5c74b02d6f0544a7dbee4935a4cf0283b5c9172e/hackathons.md#L186)):*
> ```Markdown
>     ~~~
>     CockroachDB node starting at 2017-08-17 16:42:18.303618399 +0700 ICT
>     build:      CCL v1.0.3 @ 2017/07/06 17:44:09 (go1.8.3)
> ```

Can use the liquid version tag here, too.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1840)*
<!-- Sent from Reviewable.io -->
